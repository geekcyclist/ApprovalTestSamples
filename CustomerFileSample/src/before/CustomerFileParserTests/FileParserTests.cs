using System;
using System.IO;
using System.Linq;
using Xunit;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.Windows;

namespace CustomerFileParserTests
{
    [UseReporter(typeof(BeyondCompare4Reporter))]
    [UseApprovalSubdirectory("Approvals")]
    public class FileParserTests
    {
        [Fact]
        public void CanParseCustomerData()
        {
            StringReader reader = new StringReader(MockData.SampleRows);
            var parser = new CustomerFileParser.FileParser();
            
            var customers = parser.Parse(reader);

            Assert.True(customers.Any());
        }

        [Fact(Skip = "Approval Test Stub")]
        public void ApproveParsedCustomerData()
        {
            // TODO: Implement approval of the a customer collection
            // do - create a parser and parse the file
            // verify - approve the collection
        }
    }
}
