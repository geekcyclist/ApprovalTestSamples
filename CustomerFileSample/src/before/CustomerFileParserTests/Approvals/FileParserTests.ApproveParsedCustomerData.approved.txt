Customer[0] = Id: 1256897
FirstName: Robert
MiddleName: Alan
LastName: Johnson
Suffix: Jr
Street: 7843 Bunker Hill Dr
City: Sandy
State: UT
Zip: 87074
Zip4: 1324
Age: 47

Customer[1] = Id: 1256898
FirstName: Sylvia
MiddleName: Jean
LastName: Anderson
Suffix: (null)
Street: 1821 W 6720 S Apt 7
City: West Jordan
State: UT
Zip: 84084
Zip4: (null)
Age: 31

Customer[2] = Id: 1256899
FirstName: Jennifer
MiddleName: (null)
LastName: Carlysle
Suffix: (null)
Street: 1571 Chablis Cir
City: Sandy
State: UT
Zip: 84092
Zip4: 2801
Age: 52

Customer[3] = Id: 1256900
FirstName: Jeremiah
MiddleName: James
LastName: Searle
Suffix: (null)
Street: 3167 S Patrick Dr
City: Magna
State: UT
Zip: 84044
Zip4: 3201
Age: 39

Customer[4] = Id: 1256901
FirstName: Ian
MiddleName: Patrick
LastName: O'Brian
Suffix: III
Street: 229 University Village
City: Salt Lake City
State: UT
Zip: 84108
Zip4: (null)
Age: 24

